libr/util/diff.c

```diff
@@ -358,67 +358,50 @@ R_API bool r_diff_buffers_distance_myers(RDiff *diff, const ut8 *a, ut32 la, con
 	return true;
 }
 
-R_API bool r_diff_buffers_distance_original(RDiff *d, const ut8 *a, ut32 la, const ut8 *b, ut32 lb, ut32 *distance, double *similarity) {
-	int i, j, tmin, **m;
-	ut64 totalsz = 0;
-
-	if (!a || !b || la < 1 || lb < 1)
+R_API bool r_diff_buffers_distance_original(RDiff *diff, const ut8 *a, ut32 la, const ut8 *b, ut32 lb, ut32 *distance, double *similarity) {
+	if (!a || !b)
 		return false;
 
-	if (la == lb && !memcmp (a, b, la)) {
-		if (distance != NULL)
-			*distance = 0;
-		if (similarity != NULL)
-			*similarity = 1.0;
-		return true;
-	}
-	totalsz = sizeof(int*) * (lb+1);
-	for(i = 0; i <= la; i++) {
-		totalsz += ((lb+1) * sizeof(int));
-	}
-	if (totalsz >= 1024 * 1024 * 1024) { // 1 GB of ram
-		char *szstr = r_num_units (NULL, totalsz);
-		eprintf ("Too much memory required (%s) to run distance diff, Use -c.\n", szstr);
-		free (szstr);
+	const bool verbose = diff ? diff->verbose : false;
+	const ut32 length = R_MAX (la, lb);
+	const ut8 *ea = a + la, *eb = b + lb;
+	ut32 *d, i, j;
+	// Strip prefix
+	for (; a < ea && b < eb && *a == *b; a++, b++) {}
+	// Strip suffix
+	for (; a < ea && b < eb && ea[-1] == eb[-1]; ea--, eb--) {}
+	la = ea - a;
+	lb = eb - b;
+
+	if (lb + 1 > SIZE_MAX / sizeof (ut32) || !(d = malloc ((lb + 1) * sizeof (ut32)))) {
```

Comment:
use calloc and avoid the *

Reply:
well, malloc should be fine as long as you are initializing it right after that line, and lb cant be negative and it's ok to be 0, so let's keep it like this

Reply:
Switched to `	if (sizeof (ut32) > SIZE_MAX / (lb + 1) || !(d = malloc ((lb + 1) * sizeof (ut32)))) {`

Actually I am also against using `ut32` as length types.